main.tf

```diff
@@ -1,4 +1,4 @@
-resource "azurerm_resource_group" "rg" {
+resource "azurerm_resource_group" "shared_rg" {
```

Comment:
Renaming the resource group resource is going to destroy/recreate it. I'm not quite sure what it implies for the resources it contains.

And actually, following what I said above for `vault`, given shared is already part of the context, I think `rg` is a good name.

Reply:
Agred.  I'm never sure if destroy/create happens when TF refer name changes or actual name= changes, but better safe on this one.